lib/bundler/build_metadata.rb

```diff
@@ -23,8 +23,15 @@ def self.built_at
 
     # The SHA for the git commit the bundler gem was built from.
     def self.git_commit_sha
-      @git_commit_sha ||= Dir.chdir(File.expand_path("..", __FILE__)) do
-        `git rev-parse --short HEAD`.strip.freeze
+      return @git_commit_sha if @git_commit_sha
+
+      # If Bundler has been installed without its .git directory and without a
+      # commit instance variable then we can't determine its commits SHA.
+      return "unknown" unless File.directory?(".git")
+
+      # Otherwise shell out to git.
+      @git_commit_sha = Dir.chdir(File.expand_path("..", __FILE__)) do
+        `git rev-parse --short HEAD`.strip.freeze if
```

Comment:
missing conditional

Reply:
That will teach me not to test again after cleaning up! Fixed.